src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,34 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int MIN_AGE = 18;
+    public static final int VALID_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getPassword() == null) {
+            throw new RegisterServiceImplException("Password can't be null !");
+        }
+        if (user.getLogin() == null) {
+            throw new RegisterServiceImplException("Login can't be null !");
+        }
+        if (user.getAge() < MIN_AGE && user.getAge() > 0) {
+            throw new RegisterServiceImplException("Not valid age !");
+        }
+        if (user.getAge() == 0) {
+            throw new RegisterServiceImplException("Age can't be zero !");
+        }
+        if (user.getAge() < 0) {
+            throw new RegisterServiceImplException("Age can't be nagative !");
+        }
```

Comment:
No need to have this checks. Age has already been checked here: "if (user.getAge() < MIN_AGE)"
```suggestion
```